Sources/StreamVideo/WebRTC/Screensharing/BroadcastScreenCapturer.swift

```diff
@@ -0,0 +1,118 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+import WebRTC
+import ReplayKit
+
+class BroadcastScreenCapturer: VideoCapturing {
+    
+    var bufferReader: BroadcastBufferReader?
+    var adaptedOutputFormat = false
+    
+    private var videoCapturer: RTCVideoCapturer
+    private var videoOptions: VideoOptions
+    private let videoSource: RTCVideoSource
+    private var videoCaptureHandler: StreamVideoCaptureHandler?
+    
+    init(
+        videoSource: RTCVideoSource,
+        videoOptions: VideoOptions,
+        videoFilters: [VideoFilter]
+    ) {
+        self.videoOptions = videoOptions
+        self.videoSource = videoSource
+#if targetEnvironment(simulator)
+        videoCapturer = RTCFileVideoCapturer(delegate: videoSource)
+#else
+        let handler = StreamVideoCaptureHandler(source: videoSource, filters: videoFilters, handleRotation: false)
+        videoCaptureHandler = handler
+        videoCapturer = RTCCameraVideoCapturer(delegate: handler)
+#endif
+    }
+    
+    func startCapture(device: AVCaptureDevice?) async throws {
+        guard self.bufferReader == nil else {
+            return
+        }
+        
+        guard let identifier = infoPlistValue(for: BroadcastConstants.broadcastAppGroupIdentifier),
+              let filePath = self.filePathForIdentifier(identifier)
+        else {
+            return
+        }
+        
+        let bounds = await UIScreen.main.bounds
```

Comment:
I think that accessing UIKit from a platform agnostic object like this one, is dangerous. Especially as we don't import UIKit explicitely.

Reply:
it's actually safe - `UIScreen` is a `MainActor`. That's why we need to use `await` here.